### DISCUSSION TITLE
Typegen support for @xstate/fsm

### DISCUSSION THREAD

**Discussion by schibrikov** [statelyai/xstate#3872] [Ideas] [answered]
Hi!
How hard is it to add typegen support for `@xstate/fsm` package?
`xstate` itself is huge in bundle size as to me, while `fsm` version is almost always sufficient for the basic use-cases.
I just found out that I cannot easily migrate between two, because `fsm` doesn't support typegen properties.
My guess is that config types should be almost identical except for the extra-features xstate provides, so maybe it is not that hard to support typegen in `fsm`. Honestly, I didn't read the code, but if you think this could be done I can maybe contribute into it, if you welcome the idea.

**Comment by Andarist** [comment, marked as answer]
`@xstate/fsm` is simple enough that we might be able to provide very good types for it without typegen at all. IIRC, David is revamping `@xstate/fsm` in one of the PRs - so we'll consider doing type-related changes there.

**Reply by schibrikov** [reply]
Ah, great. I thought it's kinda semi-abandoned, so that's a great news, thanks.
I have easily migrated in the end, but it's needed some effort to type actions (assign), which were typed by typegen previously.
Looking forward for David's improvements then.

**Reply by davidkpiano** [reply]
Don't worry, it's not abandoned. Working on the next version which will support better types and invokes.
